### PR TITLE
Dead Code Removal  https://github.com/gardener/machine-controller-man…

### PR DESCRIPTION
**Pull request by elankath**
Dead Code Removal  https://github.com/gardener/machine-controller-manager/issues/622

This commit removes dead and deprecated code from the machine controller manager.

**What this PR does / why we need it**:
This commit removes dead and deprecated code from the machine controller manager. The machine controller is now launched by individual providers and not by the main MCM. Old Migration code from old provider specific machine classes to harmonized machine class has also been removed..

- All provider specific types removed from `pkg/apis/machine/types.go` and api clients regenerated using `make generate`
- All migration code from provider specific classes to common MachineClass removed.
- Driver migration method removed
- Deprecated MCM flags removed.
- Removal of provider specific machine classes in `kubernetes/crds`
- Removal of provider specific machine class yamls in `kubernetes/machine_classes`
- Removal of old node drain code since this is now handled by the machine controller code present in `pkg/util/provider/drain/drain.go` which is invoked by platform specific providers.
- Some minor refactoring of routines and constants to adapt to the above.

**Which issue(s) this PR fixes**:
https://github.com/gardener/machine-controller-manager/issues/622

**Special notes for your reviewer**:
This change breaks compilation in individual provider repos which need to be updated. Removal of Driver.GenerateMachineClassForMigration needs to be conveyed to other stakeholders.

**Release note**:
```breaking operator
Removal of the following flags (and corresponding fields in associated structs): `machine-creation-timeout` `machine-drain-timeout`, `machine-pv-detach-timeout`, `machine-health-timeout=10m`, `machine-safety-apiserver-statuscheck-timeout`, `machine-safety-apiserver-statuscheck-period`, `machine-safety-orphan-vms-period`, `machine-max-evict-retries`, `node-conditions`, `bootstrap-token-auth-extra-groups`, `delete-migrated-machine-class`. The MCM no longer accepts these flags since these are options handled by the Machine Controller invoked by platform specific provider launchers.
```
```breaking developer
Deletion of `Driver.GenerateMachineClassForMigration`. Providers need to adapt to this.
```

### CODE REVIEW

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
In addition to the changes requested, kindly also remove the in-tree documentation related to how to deploy MCM 

The following changes should also be done:
Other files related to doc need to be removed :

* Delete Kubenetes/secrets and docs referencing it
* Delete Kubernetes/deployment/in-tree and docs referencing it
* Sample machine object should be updated here https://github.com/elankath/machine-controller-manager/blob/e6a1bd8a076edc94e3f3891dedcdd7dd8e754199/kubernetes/machine_objects/machine.yaml#L12

File: pkg/controller/controller_suite_test.go [hunk 1]
```diff
@@ -62,18 +62,6 @@ var (
...
-	replicas int32,
-	minReadySeconds int32,
-	statusTemplate *v1alpha1.MachineDeploymentStatus,
-	owner *metav1.OwnerReference,
-	annotations map[string]string,
-	labels map[string]string,
-) *v1alpha1.MachineDeployment {
-	return newMachineDeployments(1, specTemplate, replicas, minReadySeconds, statusTemplate, owner, annotations, labels)[0]
-}
-
```
[the remark this answers is not in the document]
Comment (elankath) [reply pkg/controller/controller_suite_test.go:76]: regarding sample at `kubernetes/machine_objects/machine.yaml` - I should just delete this as samples are now present in individual MC provider repos and sample is not needed here.

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -50,10 +50,6 @@ import (
 const (
 	maxRetries = 15
```
[the remark this answers is not in the document]
Comment (elankath) [reply pkg/controller/controller.go:51]: This is used by createWorker call at https://github.com/elankath/machine-controller-manager/blob/remdeadcode/pkg/controller/controller.go#L251.
Do you want me to replace use of this constant with a common `maxRetries` constant shared between MCM and MC controller createworker calls ?

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -68,12 +64,6 @@ func NewController(
 	pvInformer coreinformers.PersistentVolumeInformer,
```
[the remark this answers is not in the document]
Comment (elankath) [reply pkg/controller/controller.go:64]: sorry, I thought I had removed the PV/PVC informers/listers but accidentally reintroduced this after a revert. correcting this.

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -68,12 +64,6 @@ func NewController(
...
 	secretInformer coreinformers.SecretInformer,
```
[the remark this answers is not in the document]
Comment (elankath) [reply pkg/controller/controller.go]: good point, you are right, secret informer is not being used. Removed and pushed one commit that removed the pv, pvc, and secret informers/listers

File: pkg/controller/controller.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (elankath) [reply pkg/controller/controller.go]: I created a new `util/worker/worker.go` pkg and put common worker routines and constants in that package.

File: pkg/controller/controller.go [hunk 5]
```diff
@@ -19,15 +19,14 @@ package controller
...
-	"time"
-
 	machineinternal "github.com/gardener/machine-controller-manager/pkg/apis/machine"
 	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
 	machinescheme "github.com/gardener/machine-controller-manager/pkg/client/clientset/versioned/scheme"
 	machineapi "github.com/gardener/machine-controller-manager/pkg/client/clientset/versioned/typed/machine/v1alpha1"
 	machineinformers "github.com/gardener/machine-controller-manager/pkg/client/informers/externalversions/machine/v1alpha1"
 	machinelisters "github.com/gardener/machine-controller-manager/pkg/client/listers/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/worker"
+	"sync"
```
Comment (himanshu-kun) [inline pkg/controller/controller.go:29]: pls keep the "sync" package below "fmt" under import.
We usually try to follow the following grouping of imports
- any in-built libraires
- any same repo packages
- any package from third party

File: pkg/util/provider/machinecontroller/controller.go [hunk 6]
```diff
@@ -19,6 +19,7 @@ package controller
 
 import (
 	"fmt"
+	"github.com/gardener/machine-controller-manager/pkg/util/worker"
```
Comment (himanshu-kun) [inline pkg/util/provider/machinecontroller/controller.go:22]: pls keep this import in the desired location as explained in comment above

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
also pls remove the deprecated files from this file here https://github.com/elankath/machine-controller-manager/blob/ff4c295bdc2b1a2131b075795309e6f20900108a/kubernetes/deployment/out-of-tree/deployment.yaml#L24

File: pkg/controller/machine_safety.go [hunk 7]
```diff
@@ -408,258 +285,17 @@ func (c *controller) checkAndFreezeORUnfreezeMachineSets(ctx context.Context) er
...
-				}
-			}
-
-		}
-	}
-}
-
-// addMachineToSafety enqueues into machineSafetyQueue when a new machine is added
-func (c *controller) addMachineToSafety(obj interface{}) {
+// addMachineToSafetyOvershooting enqueues into machineSafetyQueue when a new machine is added
```
[the remark this answers is not in the document]
Comment (himanshu-kun) [reply pkg/controller/machine_safety.go:288]: pls address this

File: pkg/controller/controller_suite_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (himanshu-kun) [reply pkg/controller/controller_suite_test.go]: okay

File: pkg/controller/machine_safety.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (elankath) [reply pkg/controller/machine_safety.go]: Missed this. Fixed in latest commit.

File: pkg/controller/controller.go (hunk 5 above)
Comment (elankath) [reply pkg/controller/controller.go]: Cool. this formatting was done by Intellij. What I did to fix this according was go to _Settings>Editor > Code Style > Go > Imports_ and checked the "Group > Current Project Packages" and "Group stdlib imports" option. That appears to have fixed this.

File: pkg/util/provider/machinecontroller/controller.go (hunk 6 above)
Comment (elankath) [reply pkg/util/provider/machinecontroller/controller.go]: Yes, now it follows the scheme: stdlib imports, repo imports and then third party imports.

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**